pages/lessons/fundamentals/wolovim-part-1.mdx

```diff
@@ -0,0 +1,293 @@
+
+The only way for the blockchain to verify that asset was truly sent from one user to another is to use assets native to (i.e., created and governed by) that blockchain. In Ethereum, this native currency is called **ether**, and the Ethereum blockchain contains the only official record of account balances.
+
+### **A new paradigm**
+
+This new decentralized tech stack has spawned new developer tools. Such tools exist in many programming languages, but we’ll be looking through the Python lens. To reiterate: even if Python isn’t your language of choice, it shouldn’t be much trouble to follow along.
+
+Python developers that want to interact with Ethereum are likely to reach for **[Web3.py](https://web3py.readthedocs.io/)**. Web3.py is a library that greatly simplifies the way you connect to an Ethereum node, then send and receive data from it.
+
+> Note: “Ethereum node” and “Ethereum client” are often used interchangeably. In either case, it refers to the software that a participant in the Ethereum network runs. This software can read block data, receive updates when new blocks are added to the chain, broadcast new transactions, and more.
+> 
+
+Ethereum clients can be configured to be reachable by [IPC](https://en.wikipedia.org/wiki/Inter-process_communication), HTTP, or Websockets. Web3.py refers to these connection options as **providers**. You’ll want to choose one of the three providers to configure so that Web3.py knows how to communicate with your node.
+
+![https://cdn-images-1.medium.com/max/1600/1*OrElsXOF45w-AgBuezCqjQ.png](https://cdn-images-1.medium.com/max/1600/1*OrElsXOF45w-AgBuezCqjQ.png)
+
+Configure the Ethereum node and Web3.py to communicate via the same protocol, e.g., IPC in this diagram.
+
+Once Web3.py is properly configured, you can begin to interact with the blockchain. Here’s a couple of Web3.py usage examples as a preview of what’s to come:
+
+```python
+# read block data:
+w3.eth.get_block('latest')
+
+# send a transaction:
+w3.eth.send_transaction({'from': ..., 'to': ..., 'value': ...})
+```
+
+### **Installation**
+
+In this walkthrough, we’ll just be working within a Python interpreter. We won't be creating any directories, files, classes or functions.
+
+> Note: In the examples below, commands that begin with $ are intended to be run in the terminal. (Do not type the $, it just signifies the start of the line.)
+> 
+
+First, install [IPython](https://ipython.org/) for a user-friendly environment to explore in. IPython offers tab completion, among other features, making it much easier to see what’s possible within Web3.py.
+
+```bash
+$ pip install ipython
+```
+
+Web3.py is published under the name `web3`. Install the latest version like so:
+
+```bash
+$ pip install web3==6.0.0b8
```

Comment:
just a heads up: v6 stable is slated in the very near future, but `6.0.0b11` is now the latest version.